common/persistence/tasks.go

```diff
@@ -33,6 +34,7 @@ import (
 // Task is the generic interface for workflow tasks
 type Task interface {
 	GetTaskCategory() HistoryTaskCategory
+	GetTaskKey() HistoryTaskKey
```

Comment:
There's also `GetTaskID()`. Do we need both of these? If so please add comments in the interface to clarify the purpose of these methods

Reply:
GetTaskID is mostly used by the old framework